cpp/kiss_icp/cmake/CompilerOptions.cmake

```diff
@@ -22,9 +22,30 @@
 # SOFTWARE.
 function(set_global_target_properties target)
   target_compile_features(${target} PUBLIC cxx_std_17)
-  target_compile_options(${target} PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-fcolor-diagnostics>)
-  target_compile_options(${target} PRIVATE $<$<COMPILE_LANG_AND_ID:CXX,GNU>:-fdiagnostics-color=always>)
-
+  target_compile_options(
+    ${target}
+    PRIVATE # MSVC
+            $<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/W4>
+            $<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/WX>
+            $<$<COMPILE_LANG_AND_ID:CXX,MSVC>:/external:W0> # disable -isystem header warnings
+            # Clang/AppleClang
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-fcolor-diagnostics>
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Werror>
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wall>
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wextra>
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wconversion>
+            $<$<COMPILE_LANG_AND_ID:CXX,Clang,AppleClang>:-Wno-sign-conversion>
```

Comment:
Just copy-pasted from OpenVDB ;) I was trying to see what triggers. Honestly speaking, if people from OpenVDB took a different between clang and gcc flags, there might be a reasoning... I'd rather stick to it since I'm not that crazy about flags in general. But those added spitted some few more warnings here and there